### PR TITLE
Kill mesons

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Remove mesons from game and code

Also for everyone who say: "fix it not remove it"

This is not fixable. Period. The lighting system TGMC use is not compatible with the view_turf bit flags the mesons are using. And nothing currently in the lighting system can emulates what are mesons. 
And even if i spent much more time that i want of this item to try to remake it without bugs, nothing say it's even possible to remake it in the constraint of tgmc lighting system.

https://user-images.githubusercontent.com/25491240/123545790-3c157600-d75a-11eb-8aec-6534c0d6c96f.mp4

## Why It's Good For The Game

Powergamer gear, prone to multiple bugs (OW, CAS, weird behaviour with lights). This gives a advantages only to a fraction of marines, only because they are rushing/greytiding the engie vendor.

It's limited gear, such things should come from req, not "first to come have it"

## Changelog
:cl:
del: Mesons are a goner
/:cl:

### CODE REVIEW

**Comment by Jeff-jefferson-jeffery** [comment]
cope

**Comment by RenGusta** [comment]
Fix them instead of removing them, since according to some dude they have weird behaviour with new lights that isn't intended.

**Comment by BraveMole** [comment]
> Fix them instead of removing them, since according to some dude they have weird behaviour with new lights that isn't intended.

lel

**Comment by ilikesaying** [comment]
now remove xenos xray and only give them thermal vision

**Comment by RenGusta** [comment]
The video in the PR description is the one demonstrating a bug with tac binos, not the one with mesons. Don't be misleading in your PR.

I'd post the correct one but the file size is too large. 

Here's a link anyway. https://cdn.discordapp.com/attachments/504094319075131402/858691433967255582/2021-06-27_13-51-58_online-video-cutter.com.mp4

**Comment by BraveMole** [comment]
> The video in the PR description is the one demonstrating a bug with tac binos, not the one with mesons. Don't be misleading in your PR.
> 
> I'd post the correct one but the file size is too large.
> 
> Here's a link anyway. https://cdn.discordapp.com/attachments/504094319075131402/858691433967255582/2021-06-27_13-51-58_online-video-cutter.com.mp4

It's a bug due to combination of mesons + binos, not just binos.

mesons are bugged out, period

**Comment by Georgeur** [comment]
it is practically impossible to not use the bug so long as you're using binos and have mesons, the only reliable way I've found to cancel the bugged effect is by using a rail scope. 

but even with some armor like PAS or Baldur, the effect is virtually 100% night vision.
this is with PAS and mesons, without the bug. 
image

**Comment by sunofang** [comment]
Okay. And how are you gonna see where you are plasma cutting to?

**Comment by BraveMole** [comment]
> Okay. And how are you gonna see where you are plasma cutting to?

You just don't. Don't see how that would make a difference anyway

**Comment by EvilMelonTheSecond** [comment]
Just add it to req bro. 

On a serious note mesons are really valuable for engineers and the only way of using jaeger with baldur. I would much rather have a fix or a roundstart engi gear type of restriction instead of nuking it.

**Comment by Cowbin** [comment]
I agree with EvilMelonTheSecond.  Just make mesons an item in engie vendors, as they are extremely useful for engineers and plasmacutters.  If you want to lower the amount of greytiding just take them away from anywhere but engineer kit.  Any other complaints about them should be bugfixes, not removals.

**Comment by AbsoluteJerry** [comment]
Objectively, this is a bad PR. 120% not biased engineer main.

**Comment by bowlofcereal** [comment]
coping

**Comment by Katsera** [comment]
this is a massive nerf to CAS. Not even counting the bug that lets you see everything, just being unable to see in the dark kills CAS completely.

**Comment by Ieteihowmee** [comment]
How about no. What unintended affect has what had on mesons? I have not yet seen it or experienced it. Now test that bug with PAS on and no light, I bet xenos will not show up. Fix it instead of removing it.

**Comment by Ieteihowmee** [comment]
> it is practically impossible to not use the bug so long as you're using binos and have mesons, the only reliable way I've found to cancel the bugged effect is by using a rail scope.
> 
> but even with some armor like PAS or Baldur, the effect is virtually 100% night vision.
> this is with PAS and mesons, without the bug.
> image

Binos cause the issue? Interesting. I was thinking in the video brave posted that he was able to see the runner because of the baulder module. Since you CAN see them in some dark spots with mesons but only as far as light will emit and a few tiles out. Interesting. Still, should just fix it instead of remove it. More marine nerfs hold dang.

**Comment by Katsera** [comment]
also: salt pr

**Comment by Alexeivichi** [comment]
The community genuinely thinks this is a bad PR. Even worse than MJP's remove valk PR.

**Comment by BraveMole** [comment]
> this is a massive nerf to CAS. Not even counting the bug that lets you see everything, just being unable to see in the dark kills CAS completely.

Ah yes, this is exactly the kind of stuff that is totally not intented and broken.

**Comment by MLGTASTICa** [comment]
> this is a massive nerf to CAS. Not even counting the bug that lets you see everything, just being unable to see in the dark kills CAS complet

Dude , CAS is not meant to be able to be used through the dark , its an actual balance change , you need to light up a area if you want to use CAS

**Comment by Katsera** [comment]
> 
> 
> > this is a massive nerf to CAS. Not even counting the bug that lets you see everything, just being unable to see in the dark kills CAS complet
> 
> Dude , CAS is not meant to be able to be used through the dark , its an actual balance change , you need to light up a area if you want to use CAS

Well then CAS is useless if that's the case.

**Comment by AbsoluteJerry** [comment]
Just fix mesons instead of removing them completely

**Comment by AnCopper** [comment]
> Just fix mesons instead of removing them completely

It's a shame that you can't read.

**Comment by ZahaSanko** [comment]
cope

**Comment by lovegreenstuff** [comment]
It's weird that you call it limited gear and say it should be in req, but instead of just removing it from every shipside instance and adding it to req, you just delete it in its entirety. 

Also, there's tons of limited supplies that are first come, first serve. Pills and certain guns and equipment come to mind.
Powergaming isn't a reason to delete something in a TDM where winning is the entire goal. 

What do I know? I'm just some dingus playing space game.
It just seems like another silly Bravemole PR with very questionable intentions.

**Comment by AbsoluteJerry** [comment]
> > Just fix mesons instead of removing them completely
> 
> It's a shame that you can't read.

Explain to me what I missed or shove those words right back into your oxygen wasting mouth.

**Comment by Captiajoe** [comment]
The mesons give you view of the dark tiles for sure, however you cant see anything that is on that tile, xenos or item until a light is in range of it, try to see a xenos in your corner of your screen that is bathed in darkness, than when it gets close to your armor light you will than see it on mesons and without it, unless that was changed.

**Comment by Georgeur** [comment]
It does effectively give you night vision, and if you can't hold two separate beliefs at once, like mesons are unbalanced/bugged, but also we should have something to look for cave openings and building layouts. 
Try to use a few braincells, they're not wrong that mesons would be god awful to "fix".

**Review by SplinterGP — CHANGES_REQUESTED** [review]
you probably said nuuh, but fix the bugs not remove it, its completely possible to fix them as they were sometimes already fixed before.

**Comment by bowlofcereal** [comment]
there is a distinct massive amount of copium occurring in this comment section

**Comment by Ragecrabbo** [comment]
Medhud is better anyway

**Comment by Georgeur** [comment]
I dont want some unrobust people talking the chat, it gives you NV, end of. 
Don't agree with the pr but it should be TM'd until we can find a solution.

**Comment by usnpeepoo** [comment]
based pr, people crying about mesons being remove when they can't do the same prep routine of tiding mesons like they did with MJP's valk removal pr

**Comment by Ieteihowmee** [comment]
> > Just fix mesons instead of removing them completely
> 
> It's a shame that you can't read.

Keep in mind. I am pretty sure that was an updated description post fact. But nice try, trying to be smart go you!

**Comment by TiviPlus** [comment]
Next person to comment off topic means noone gets to comment but me

**Comment by TiviPlus** [comment]
Next person to comment off topic means noone gets to comment but me

File: code/modules/clothing/glasses/meson.dm [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-	name = "optical meson scanner"
-	desc = "Used to shield the user's eyes from harmful electromagnetic emissions, also used as general safety goggles. Not adequate as welding protection."
-	icon_state = "meson"
-	item_state = "meson"
-	deactive_state = "degoggles_meson"
-	actions_types = list(/datum/action/item_action/toggle)
-	toggleable = 1
-	darkness_view = 2
-	vision_flags = SEE_TURFS
-	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
```
Comment (TiviPlus) [inline code/modules/clothing/glasses/meson.dm:14]: why arent you just removing the lighting alpha changes so you can only see as far as the light shines

**Pull request closed without merging**